From stable-bounces@linux.kernel.org Sun Apr 23 15:45:53 2006
Message-Id: <200604232245.k3NMj7Zw000433@shell0.pdx.osdl.net>
To: dsd@gentoo.org, stable@kernel.org, mm-commits@vger.kernel.org
From: akpm@osdl.org
Date: Sun, 23 Apr 2006 15:43:59 -0700
Cc: 
Subject: tipar oops fix

From: Daniel Drake <dsd@gentoo.org>

If compiled into the kernel, parport_register_driver() is called before the
parport driver has been initalised.

This means that it is expected that tp_count is 0 after the
parport_register_driver() call() - tipar's attach function will not be
called until later during bootup.

Signed-off-by: Daniel Drake <dsd@gentoo.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/tipar.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- linux-2.6.16.11.orig/drivers/char/tipar.c
+++ linux-2.6.16.11/drivers/char/tipar.c
@@ -515,7 +515,7 @@ tipar_init_module(void)
 		err = PTR_ERR(tipar_class);
 		goto out_chrdev;
 	}
-	if (parport_register_driver(&tipar_driver) || tp_count == 0) {
+	if (parport_register_driver(&tipar_driver)) {
 		printk(KERN_ERR "tipar: unable to register with parport\n");
 		err = -EIO;
 		goto out_class;
